code/__HELPERS/paths/sssp.dm

```diff
@@ -0,0 +1,300 @@
+
+	// We're gonna build a pathfind datum from our settings and set it running
+	var/datum/pathfind/sssp/based_off_us = new()
+
+	based_off_us.setup_from_canpass(pass_info, start, new_range, pass_space, avoid, list(await))
+	based_off_us.working_queue = next_closest.Copy()
+	based_off_us.working_distances = working_distances.Copy()
+	based_off_us.working_index = working_index
+	if(!SSpathfinder.run_pathfind(based_off_us))
+		return FALSE
+
+	expanding = TRUE
+	UNTIL(length(hand_around))
+	var/datum/path_map/return_val = hand_around[1]
+	if(!istype(return_val, /datum/path_map)) // It's trash, we've failed and need to clear away
+		return FALSE
+	copy_from(return_val)
+	expanding = FALSE
+	return TRUE
+
+/datum/path_map/proc/sanity_check()
+	for(var/index in 1 to length(distances))
+		var/turf/next_turf = next_closest[index]
+		var/list/path = get_path_from(next_turf)
+		if(length(path) != distances[index] + 1)
+			stack_trace("[next_turf] had a distance of [length(path)] instead of the expected [distances[index]]")
+		if(path.Find(next_turf) != 1)
+			stack_trace("Starting turf [next_turf] was not the first entry in its list (instead it's at [path.Find(next_turf)])")
+		path = get_path_to(next_turf)
+		if(length(path) != distances[index] + 1)
+			stack_trace("[next_turf] had a distance of [length(path)] instead of the expected [distances[index]]")
+		if(path.Find(next_turf) != length(path))
+			stack_trace("Starting turf [next_turf] was not the last entry in its list (instead it's at [path.Find(next_turf)])")
+
+/// Single source shortest path
+/// Generates a flow map of a reachable turf -> the turf next closest to the map's center
+/datum/pathfind/sssp
+	/// Ever expanding list of turfs to visit/visited, associated with the turf that's next closest to them
+	var/list/working_queue
+	/// List of distances, each entry mirrors an entry in the working_queue
+	var/list/working_distances
+	/// Our current position in the working queue
+	var/working_index
+
+/datum/pathfind/sssp/proc/setup(atom/movable/caller, list/access, turf/center, max_distance, simulated_only, turf/avoid, list/datum/callback/on_finish)
+	src.pass_info = new(caller, access)
+	src.start = center
+	src.max_distance = max_distance
+	src.simulated_only = simulated_only
+	src.avoid = avoid
+	src.on_finish = on_finish
+
+/datum/pathfind/sssp/proc/setup_from_canpass(datum/can_pass_info/info, turf/center, max_distance, simulated_only, turf/avoid, list/datum/callback/on_finish)
+	src.pass_info = info
+	src.start = center
+	src.max_distance = max_distance
+	src.simulated_only = simulated_only
+	src.avoid = avoid
+	src.on_finish = on_finish
+
+/datum/pathfind/sssp/start()
+	. = ..()
+	if(!.)
+		return .
+	working_queue = list()
+	working_distances = list()
+	working_queue[start] = FLOW_PATH_END
+	working_distances += 0
+	working_index = 0
+	return TRUE
+
+/datum/pathfind/sssp/search_step()
+	. = ..()
+	if(!.)
+		return
```

Comment:
```suggestion
		return .
```

we already do this on the above proc so may as well be consistent

Reply:
yes honey